src/formats/onenote.ts

```diff
@@ -357,17 +385,108 @@ export class OneNoteImporter extends FormatImporter {
 		return pageElement;
 	}
 
+	/**
+	 * Returns a filesystem path for any OneNote entity like sections or notes
+	 * Paths are returned in the following format:
+	 * (Export folder)/Notebook/(possible section groups)/Section/(possible pages with a higher level) 
+	 */
+	getEntityPath(entityID: string, currentPath: string, parentEntity?: OnenoteEntityHierarchyModel | undefined): string | null {
```

Comment:
I think this function is trying to do too much and has too much casting. Maybe it would be simpler and safer if there were different helper functions for getting a path depending on the parent entity type.

Reply:
Hmm, your concern about this function is definitely true, however the OneNote importer is already pretty big (more functions than the most other importers), and I'm worried that splitting into multiple functions make the code harder to understand. Or perhaps, we could split some functions off into into multiple files like the Apple Notes importer. Please let me know what you think.

Reply:
I don't think splitting into functions is necessarily going to make the code bigger. And removing the cases where we are stepping around type safety will make the code easier to understand and safer.

Reply:
I like how this has been split up into separate functions, thank you! I'm wondering now if this function is actually needed at all. There are 4 places where we call it, each time with a different parentEntity type. Let's just call the different helper functions directly.